sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,120 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=3, normalize=True,
+                         sample_weight=None):
+    """Top-k Accuracy multiclass classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top `k` labels predicted (ranked by predicted scores). Note that the
+    multilabel and multioutput cases aren't covered here.
+
+    Read more in the :ref:`User Guide <top_k_accuracy_score>`
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels. Expected to be a sequence of int in the range
+        (0, n_classes - 1).
```

Comment:
```suggestion
        [0, n_classes - 1].
```

`(a, b)` is actually what some of us would write `]a, b[`.

Otherwise we can use "a sequence of ints in `range(n_classes)`"